pkg/client/generic_client.go

```diff
@@ -0,0 +1,45 @@
+package client
+
+import (
+	monitoringclientset "github.com/coreos/prometheus-operator/pkg/client/versioned"
+	kubeclientset "k8s.io/client-go/kubernetes"
+	rest "k8s.io/client-go/rest"
+)
+
+type GenericClientset struct {
+	KubeClient   kubeclientset.Interface
+	MonitoringClient monitoringclientset.Interface
+}
+
+// NewForConfig creates a new Clientset for the given config.
+func newForConfig(c *rest.Config) (*GenericClientset, error) {
+	kubeClient, err := kubeclientset.NewForConfig(c)
+	if err != nil {
+		return nil, err
+	}
+	monitoringClient, err := monitoringclientset.NewForConfig(c)
```

Comment:
This is essentially all you need to connect to the monitoring client, so I don't think this helper function is really needed.